fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/FenixOverlay.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.debugsettings.store.DebugDrawerAction
+import org.mozilla.fenix.debugsettings.store.DebugDrawerStore
+import org.mozilla.fenix.debugsettings.store.DrawerStatus
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Overlay for presenting Fenix-wide debugging content.
+ */
+@Composable
+fun FenixOverlay() {
+    val debugDrawerStore = remember { DebugDrawerStore() }
```

Comment:
Are there any perf considerations for remembering a complex type like this?

Reply:
Only in the positive sense that it'll help save on recompositions. Other than having `DebugDrawerStore` as a parameter for `FenixOverlay` and passed-in here via `HomeActivity`, this is the only way to ensure that store does not get nuked during a random recomp. 

Thinking aloud though, this may need to be re-examined in the lifecycle case though (like when the device is rotated) 🤔